dev/com.ibm.ws.microprofile.reactive.messaging_fat/fat/src/com/ibm/ws/microprofile/reactive/messaging/fat/kafka/partitions/LivePartitionTestServlet.java

```diff
@@ -0,0 +1,156 @@
+import org.junit.Test;
+
+import com.ibm.ws.microprofile.reactive.messaging.fat.kafka.framework.AbstractKafkaTestServlet;
+import com.ibm.ws.microprofile.reactive.messaging.fat.kafka.partitions.LivePartitionTestBean.AckStatus;
+import com.ibm.ws.microprofile.reactive.messaging.fat.kafka.partitions.LivePartitionTestBean.ReceivedMessage;
+
+/**
+ * Tests partition rebalancing in something more akin to a live environment
+ * <p>
+ * In particular, we want to test rebalancing while the application is actively processing messages to ensure that none are lost and they're handled correctly.
+ */
+@WebServlet("/LivePartitionTest")
+public class LivePartitionTestServlet extends AbstractKafkaTestServlet {
+
+    private static final long serialVersionUID = 1L;
+
+    public static final String APP_GROUPID = "kafka-live-partition-test-group";
+
+    @Resource
+    private ManagedExecutorService executor;
+
+    @Inject
+    private LivePartitionTestBean bean;
+
+    @Test
+    public void testLivePartitionAssignment() throws Exception {
+
+        List<String> sentMessages = new ArrayList<>();
+
+        Map<String, Object> producerConfig = new HashMap<>();
+        producerConfig.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, getKafkaBootstrap());
+        producerConfig.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+        producerConfig.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+
+        // Load messages into topic
+        try (KafkaProducer<String, String> producer = new KafkaProducer<>(producerConfig)) {
+            for (int partition = 0; partition < 2; partition++) {
+                for (int message = 0; message < 100; message++) {
+                    String value = partition + "-" + message;
+                    ProducerRecord<String, String> record = new ProducerRecord<String, String>(LivePartitionTestBean.CHANNEL_IN, partition, null, value);
+                    producer.send(record);
+                    sentMessages.add(value);
+                }
+
+                // Add a sentinal message to the end of each partition
+                String value = partition + "-" + LivePartitionTestBean.FINAL_MESSAGE_NUMBER;
+                ProducerRecord<String, String> record = new ProducerRecord<String, String>(LivePartitionTestBean.CHANNEL_IN, partition, null, value);
+                producer.send(record);
+                sentMessages.add(value);
+            }
+        }
+
+        Map<String, Object> consumerConfig = new HashMap<>();
+        consumerConfig.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, getKafkaBootstrap());
+        consumerConfig.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+        consumerConfig.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+        consumerConfig.put(ConsumerConfig.GROUP_ID_CONFIG, APP_GROUPID);
+
+        // Sleep
+        Thread.sleep(700);
```

Comment:
constant?

Reply:
is there any other way than using a sleep?

Reply:
Not really, it's trying to simulate a live environment where other consumers can join and leave the group at random. I'm open to other suggestions.

The design of the test is just to provoke a bunch of things to happen and then check that the runtime handled whatever happened in a valid way.

Reply:
I don't think it's actually helpful to turn this into a constant. The length of time here is pretty arbitrary. It's more important for someone reading the test to be able to see at a glance how long it's sleeping for.